configurator.go

```diff
@@ -73,38 +82,114 @@ func (c *Config) Parse() (*Config, diag.Diagnostics) {
 
 	// default filename to 'config' if not provided.
 	if c.FileName == "" {
-		c.FileName = "config"
+		c.FileName = DEFAULT_CONFIG_FILENAME
 	}
 	c.parsed = nil
 
-	// Process OS application directory configuration files.
-	c, diags = processFileConfig(diags, diag.ComponentGlobalFile, c)
+	// todo: make the following processX functions private methods
+
+	c, diags = processFileFlagConfig(diags, c)
```

Comment:
attempting to override `c` here won't change the parent, and the new values will only be available to the children who usee the returned config.

These functions should be changed to methods in order to affect the parent properly.